client/src/model/data_layer.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Class representing a data layer on the map.
+ */
+export class DataLayer {
```

Comment:
Can this be an interface. It seems like in typescript you can just do interface DataLayer {name: string} and you can instantiate it? Pls correct me if I'm wrong bc I'm a total noob to typescript. I would try that or just name:string without the getter since it's not private right now anyway and doesn't seem necessary putting it behind getters?

Reply:
Great point - I think the downside is it won't have a constructor, it is more like defining a type for js objects, however for now you are right we don't need additional functionality. 

Depending on how we want to store data here it may make sense to convert it to class or implement the interface as a class in the future but for now this is fine.